static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -252,55 +263,84 @@ const STATES = {
     PAUSED: onPlayerPaused,
 };
 
-const getIFrameBehaviour = (atomId: string): IFrameBehaviour => {
-    const isAutoplayBlockingPlatform = () => isIOS() || isAndroid();
+const getIFrameBehaviourConfig = (
+    iframe: HTMLIFrameElement
+): IFrameBehaviourConfig => {
+    const isAutoplayBlockingPlatform = isIOS() || isAndroid();
 
-    const isInternalReferrer = (): boolean => {
+    const isInternalReferrer = ((): boolean => {
         if (config.get('page.isDev')) {
             return document.referrer.indexOf(window.location.origin) === 0;
         }
 
         return document.referrer.indexOf(config.get('page.host')) === 0;
-    };
+    })();
 
-    const isMainVideo = (): boolean =>
-        (players[atomId].iframe &&
-            !!players[atomId].iframe.closest(
-                'figure[data-component="main video"]'
-            )) ||
+    const isMainVideo =
+        (iframe && !!iframe.closest('figure[data-component="main video"]')) ||
         false;
 
-    const flashingElementsAllowed = (): boolean =>
-        accessibilityIsOn('flashing-elements');
+    const flashingElementsAllowed = accessibilityIsOn('flashing-elements');
 
-    const isVideoArticle = (): boolean =>
+    const isVideoArticle =
         config.get('page.contentType', '').toLowerCase() === 'video';
 
-    const isFront = () => config.get('page.isFront');
+    const isFront = config.get('page.isFront');
     const isUSContent =
         config.get('page.productionOffice', '').toLowerCase() === 'us';
 
     const isPaidContent = config.get('page.isPaidContent');
+
+    return {
+        isAutoplayBlockingPlatform,
+        isInternalReferrer,
+        isMainVideo,
+        flashingElementsAllowed,
+        isVideoArticle,
+        isFront,
```

Comment:
Off-topic... 🙈 🙊 
I am surprised this is passing type checking? The return type of `config.get` is `any` so there is no guarantee (as far as flow knows) that `isFront` will be a `boolean`

Reply:
True. I think the only one that isn't guaranteed to be a boolean is `const isFront = config.get('page.isFront');` and changing that to `const isFront = config.get('page.isFront', false);` should cover that from a developer point of view at least.